src/App.jsx

```diff
@@ -1,37 +1,198 @@
     const newTodo = {
       id: Date.now(),
       title: title,
       isCompleted: false
     }
-    setTodoList([...todoList, newTodo])
+    const payload = {
+      records: [
+        {
+          fields: {
+            title: newTodo.title,
+            isCompleted: newTodo.isCompleted,
+          },
+        },
+      ],
+    };
+    const options = {
+      method: 'POST',
+      headers: {
+        Authorization: token,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(payload),
+    };
+    try {
+      setIsSaving(true)
+      const resp = await fetch(url, options)
+      if (!resp.ok) {
+        new Error(resp.statusText)
+      }
+      const {records} = await resp.json()
+      const { id, fields } = records[0];
+      const savedTodo = {
+        id,
+        ...fields,
+      };
+      if (!records[0].fields.isCompleted) {
+        savedTodo.isCompleted = false;
+      }
+      setTodoList([...todoList, savedTodo]);
+    } catch (error) {
+      setErrorMessage(error.message)
+      console.error(error.message)
+    } finally {
+      setIsSaving(false)
+    }
   }
-  function completeTodo(id) {
-    const updatedTodo = todoList.map(ele => {
-      if (ele.id === id) return {...ele, isCompleted: true}
-      return ele
+  async function completeTodo(id) {
+    const completeTodo = todoList.find((todo) => todo.id === id);
+    const updateTodoList = todoList.map((todo) => {
+      if (todo.id === id) {
+        return { ...todo, isCompleted: !todo.isCompleted };
+      }
+      return todo;
     });
-    setTodoList(updatedTodo);
+    setTodoList(updateTodoList);
+    setIsSaving(true);
+
+    const payload = {
+      records: [
+        {
+          id: completeTodo.id,
+          fields: {
+            title: completeTodo.title,
+            isCompleted: true,
+          },
+        },
+      ],
+    };
+    const options = {
+      method: 'PATCH',
+      headers: {
+        Authorization: token,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(payload),
+    };
+    try {
+      const resp = await fetch(url, options);
+      if (!resp.ok) {
+        throw new Error(resp.statusText);
+      }
+    } catch (error) {
+      console.log(error);
+      setErrorMessage(`${error.message}. Reverting todo completion...`);
+      const revertedTodos = todoList.map((todo) =>
+        todo.id === completeTodo.id ? completeTodo : todo
+      );
+      setTodoList([...revertedTodos]);
+    } finally {
+      setIsSaving(false);
+    }
   }
-  function updateTodo(editedTodo) {
-    const updatedTodo = todoList.map(ele => {
-      if (ele.id === editedTodo.id) return {...editedTodo}
-      return ele
-    });
+  async function updateTodo(editedTodo) {
+    const originalTodo = todoList.find((todo) => todo.id === editedTodo.id);
+    const updatedTodoList = todoList.map((todo) =>
+      todo.id === editedTodo.id ? editedTodo : todo
+    );
+    setTodoList(updatedTodoList);
```

Comment:
You're calling `setTodoList(updatedTodoList);` twice here. I would remove of them.